org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/TypeHierarchyCommand.java

```diff
@@ -230,6 +230,9 @@ private void resolve(TypeHierarchyItem item, IMember member, IMethod targetMetho
 			List<TypeHierarchyItem> childrenItems = new ArrayList<>();
 			IType[] children = typeHierarchy.getSubtypes(type);
 			for (IType childType : children) {
+				if (monitor.isCanceled() || resolve <= 0) {
```

Comment:
You don't need to check for `resolve <= 0` here, and below. We do it first thing when we enter the method. Otherwise, I think the locations of cancellation checking are good.